src/Microsoft.Extensions.Configuration/ConfigurationBuilder.cs

```diff
@@ -58,6 +59,31 @@ public IConfigurationBuilder Add(IConfigurationProvider provider, bool load)
         }
 
         /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public IConfigurationBuilder AddConfiguration(IConfiguration config)
+        {
+            Add(new ChainedConfigurationProvider(config));
+            return this;
+        }
+
+        /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <param name="key">The key of the root of the configuartion to inculde.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public IConfigurationBuilder AddConfiguration(IConfiguration config, string key)
```

Comment:
It seems like low value sugar, but it was requested for in the initial issue so I just added it.